core/jobs.py

```diff
@@ -210,7 +213,8 @@ def register_completion(cls, job_id, output_list):
 
         model.status_code = STATUS_CODE_COMPLETED
         model.time_finished_msec = utils.get_current_time_in_millisecs()
-        model.output = cls._compress_output_list(output_list)
+        model.output = cls._compress_output_list(
+            output_list, test_only_max_output_len_chars=max_output_len_chars)
```

Comment:
Perhaps drop the test_only from here, and move _max_output_len_chars to this method instead. At the outset of the method you can say _max_output_len_chars = (max_output_len_chars or 900000) (or something similar)